src/color-picker/color-picker.js

```diff
@@ -50,22 +59,39 @@ const Self = class ColorPicker extends ColorElement {
 			}
 			this.color = this._el.swatch.color;
 		}
+		else if (this._el.space_picker.contains(source) || this._slots.space_picker.assignedElements().includes(source)) {
+			this.space = event.target.value;
+			this.color = this.color.to(this.space);
+		}
 
 		this.dispatchEvent(new event.constructor(event.type, {...event}));
 	}
 
+	get spaceResolved () {
+		return this._el.space_picker.selectedSpace;
```

Comment:
What would this return if we don’t have a space picker? The space picker is interchangeable UI, we shouldn't depend on it for resolving the color space.

Reply:
It looks like the default `<space-picker>` is always there (even if we have some slotted content instead of it). Its value is always in sync with the `space` property. So, the resolved color space should always be correct.

<img width="879" alt="image" src="https://github.com/user-attachments/assets/ba741530-2257-4e2d-a06e-600cae41ebfb">

What am I missing?

Reply:
Hmmmm. It still seems like a hack but ok